lib/statsd.js

```diff
@@ -71,15 +72,24 @@ function makeStatsD(options, logger) {
         statsd = new StatsD(statsdOptions);
     }
 
-    statsd._children = [];
     // Support creating sub-metric clients with a fixed prefix. This is useful
     // for systematically categorizing metrics per-request, by using a
     // specific logger.
     statsd.makeChild = (name) => {
         const childOptions = Object.assign({}, options);
-        childOptions._prefix = statsdOptions.prefix + normalizeName(name);
-        const child = makeStatsD(childOptions, logger);
-        statsd._children.push(child);
+        childOptions.prefix = statsdOptions.prefix + normalizeName(name);
+
+        const child = statsd.childClient(childOptions);
+
+        // Attach normalizeName to make the childClient instance backwards-compatible
+        // with the previously used StatsD instance
+        child.normalizeName = normalizeName;
+
+        // Also add a small utility to send a delta given a startTime
+        child.endTiming = function endTiming(names, startTime, samplingInterval) {
+            return this.timing(names, Date.now() - startTime, samplingInterval);
+        };
```

Comment:
This code is duplicated on line 99. We can just make it a const endTiming and then bind it to an appropriate 'this'.

Alternatively event better, we can make a function that will apply the mixin to the original hot-shots statsd and share it for both the root and the child.

Reply:
Added mixin